code/game/objects/items/devices/radio/headset.dm

```diff
@@ -358,15 +358,15 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 
 /obj/item/radio/headset/screwdriver_act(mob/living/user, obj/item/tool)
 	user.set_machine(src)
-	if(keyslot || keyslot2)
+	if(keyslot || keyslot2 && !istype(keyslot2, /obj/item/encryptionkey/ai)) //Check to make sure you cant remove the AI key from core 
```

Comment:
I don't really like this hardcode istype, perhaps there should be a trait or variable that denotes an encryption key as unremovable ? If i'm understanding the purpose of the code correctly

Reply:
Purpose is really only to make sure that players cant remove specifically the AI key, since this one has all station channels and ai private.
If i have understood it correctly, its a key that IC doesnt exist and should not be able to end up in a players hand.

Not sure if there is a better way to do it, but I do welcome feedback and ideas.

Reply:
I understand your purpose but I would rather we had a variable or something to denote `this key should never be removed from headsets as players should not have their hands on it` 
...Rather than checking for this type directly

Reply:
That actually makes a lot of sense and shouldn't be too difficult to work in to set if a key can or cant be removed.
I will see what magic I can work up tomorrow.